### PR TITLE
Fix ZStream#debounce interruption issue

### PR DESCRIPTION
**Pull request by luis3m**
Relates to #3740

### CODE REVIEW

**Comment by luis3m** [comment]
Actually, I think only the `cause.interrupted` is needed here. The other two cases require the fiber to finish. Will update in a few minutes.

**Comment by luis3m** [comment]
Done

**Comment by luis3m** [comment]
I will change `tapError` with `onInterrupt`

**Comment by luis3m** [comment]
Done

**Comment by luis3m** [comment]
@iravid this is ready for review

File: streams/shared/src/main/scala/zio/stream/ZStream.scala [hunk 1]
```diff
@@ -2734,7 +2734,13 @@ abstract class ZStream[-R, +E, +O](val process: ZManaged[R, Nothing, ZIO[R, Opti
...
             case Done =>
               Pull.end
-          }
+          }.onInterrupt(
+            ref.get.flatMap {
+              case Previous(fiber) => fiber.interrupt
+              case Current(fiber)  => fiber.interrupt
+              case _               => ZIO.unit
+            }
+          )
```
Comment (iravid) [inline streams/shared/src/main/scala/zio/stream/ZStream.scala:2743]: Sorry for this PR dragging out. It's probably better for this finalizer to be attached to the construction of the `Ref`; this way the fiber is interrupted even if the stream is interrupted "in-between" pulls. What do you think?

Comment (luis3m) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala:2743]: @iravid I think you mean:
```scala
Ref.make[State](NotStarted).toManaged {
  _.get.flatMap {
    case Previous(fiber) => fiber.interrupt
    case Current(fiber)  => fiber.interrupt
    case _               => ZIO.unit
  }
}
```
Don't you?

Comment (iravid) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala:2743]: Yes, exactly

Comment (luis3m) [reply streams/shared/src/main/scala/zio/stream/ZStream.scala]: @iravid makes sense, done. Also added a new test case for the fiber interruption.

**Pull request merged**